lua/nvim_lsp/nimls.lua

```diff
@@ -0,0 +1,43 @@
+local configs = require 'nvim_lsp/configs'
+local util = require 'nvim_lsp/util'
+
+configs.nimls = {
+  default_config = {
+    cmd = {"nimlsp",};
+    filetypes = {'nim'};
+    log_level = vim.lsp.protocol.MessageType.Warning;
+    settings = {};
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or vim.loop.os_homedir()
+    end;
+  };
+  docs = {
+    description = [[
+https://github.com/PMunch/nimlsp
+`nimlsp` can be installed via `:LspInstall nimls` or by yourself the `nimble` package manager:
+```sh
+nimble install nimlsp
+```
+    ]];
```

Comment:
Would you add these lines?
```
    default_config = {
      root_dir = [[root_pattern(".git") or os_homedir]];
    };
```

Reply:
These lines are already here:

https://github.com/neovim/nvim-lsp/pull/122/files#diff-ef85a4b4f95fcf92bafc2693e4a92f6dR9-R11

Should I add them again in the `install` function?

Reply:
Yes, this is to generating README.md.
If you don't add those lines, it'll be `root_dir = <function 1>`.

Reply:
I added 

```
    default_config = {
      root_dir = [[root_pattern(".git") or os_homedir]];
    };
```

to the docs block in 14f487f21ef97fcf923e304d8eaa76c872f276bc.